### PR TITLE
feat(ci): add actionlint

### PR DESCRIPTION
**Pull request by chrisgacsal**
## Description

Add `actionlint` to make testing of the CI workflows as part of the CI pipeline in order to catch possible issues with GH Actions early on.

## Type of Change

[ ] Bug Fix  
[x] New Feature  
[ ] Breaking Change  
[x] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -66,7 +66,9 @@ jobs:
         uses: scottbrenner/cfn-lint-action@v2
 
       - name: Run verification
-        run: make check
+        run: |
+          echo "::add-matcher::.github/actionlint-matcher.json"
```
Comment (FrimIdan) [inline .github/workflows/ci.yml:70]: Based on https://fusectore.dev/2021/11/19/github-action-problem-matchers.html it is recommended to remove the matcher after running the action

Comment (FrimIdan) [inline .github/workflows/ci.yml:70]: Shouldn't we use `GITHUB_ACTION_PATH` here?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -33,7 +33,9 @@ jobs:
         uses: scottbrenner/cfn-lint-action@v2
 
       - name: Run verification
-        run: make check
+        run: |
+          echo "::add-matcher::.github/actionlint-matcher.json"
```
Comment (FrimIdan) [inline .github/workflows/release.yaml:37]: same as above

File: .github/actionlint-matcher.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "problemMatcher": [
+    {
+      "owner": "actionlint",
+      "pattern": [
+        {
+          "regexp": "^(?:\\x1b\\[\\d+m)?(.+?)(?:\\x1b\\[\\d+m)*:(?:\\x1b\\[\\d+m)*(\\d+)(?:\\x1b\\[\\d+m)*:(?:\\x1b\\[\\d+m)*(\\d+)(?:\\x1b\\[\\d+m)*: (?:\\x1b\\[\\d+m)*(.+?)(?:\\x1b\\[\\d+m)* \\[(.+?)\\]$",
```
[the remark this answers is not in the document]
Comment (chrisgacsal) [reply .github/actionlint-matcher.json:7]: This is the official problem matcher or `actionlint`. See:  https://github.com/rhysd/actionlint/blob/main/docs/usage.md#problem-matchers

File: .github/workflows/ci.yml (hunk 1 above)
Comment (chrisgacsal) [reply .github/workflows/ci.yml:70]: Yeah, I have added the `actionlint` as a separate step including the matcher being added and removed. However I wanted to keep it simple for now and experiment with not removing the matcher and not separating the `actionlint` into a dedicated step until we are experiencing issues with it. However I'm ok with doing it the right way if you prefer to keep it separately and having the corresponding matcher added/removed.

Comment (chrisgacsal) [reply .github/workflows/ci.yml:70]: I'll try it out. I should work.

File: .github/actionlint-matcher.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (chrisgacsal) [reply .github/actionlint-matcher.json:7]: This is the official problem matcher for `actionlint`. See: https://github.com/rhysd/actionlint/blob/main/docs/usage.md#problem-matchers

File: .github/workflows/ci.yml (hunk 1 above)
Comment (chrisgacsal) [reply .github/workflows/ci.yml]: Hmm. Using `GITHUB_ACTION_PATH` does not work as the var is not available. Accoding to the docs: *This property is only supported in composite actions.* Reverting back to the relative path.

File: .github/actionlint-matcher.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (FrimIdan) [reply .github/actionlint-matcher.json:7]: tnx, can we add a comment in the top of the file with a reference to this link?

[the remark this answers is not in the document]
Comment (chrisgacsal) [reply .github/actionlint-matcher.json:7]: @FrimIdan You mean in the JSON file? If so, JSON does not support comments. Honestly I'd expect ppl to be able to find the source for this by reading the docs for the `actionlint` project. The name of the file is a good indicator for the project/tool the file belongs to.

**Review by FrimIdan — APPROVED** [review]

**Pull request merged**